compiler/noirc_frontend/src/tests.rs

```diff
@@ -1724,11 +1721,6 @@ fn numeric_generic_as_param_type() {
         errors[1].0,
         CompilationError::TypeError(TypeCheckError::TypeKindMismatch { .. }),
     ));
-    // Error from the return type
```

Comment:
We do not error for the return type anymore? I would still expect this error.

Reply:
I found and fixed two bugs from this:
- It was actually the error from the parameter type (added tests to distinguish the cases)
- The parameter types were being defaulted to `Kind::Any`, but only `Kind::Normal` should be allowed